firebase-installations/src/test/java/com/google/firebase/installations/FirebaseInstallationsTest.java

```diff
@@ -453,6 +454,40 @@ public void testReadToken_withJsonformatting() {
     assertThat(iidStore.readToken(), equalTo("thetoken"));
   }
 
+  @Test
+  public void testFidListener_fidChanged_successful() throws Exception {
+    when(mockIidStore.readIid()).thenReturn(null);
+    when(mockIidStore.readToken()).thenReturn(null);
+    when(mockBackend.createFirebaseInstallation(
+            anyString(), anyString(), anyString(), anyString(), any()))
+        .thenReturn(
+            TEST_INSTALLATION_RESPONSE
+                .toBuilder()
+                .setUri("/projects/" + TEST_PROJECT_ID + "/installations/" + TEST_FID_2)
+                .setFid(TEST_FID_2)
+                .build());
+
+    FakeFidListener fidListener = new FakeFidListener();
+    // Register the FidListener
+    firebaseInstallations.registerFidListener(fidListener);
```

Comment:
Please consider adding a test for removing a registration as well.

Reply:
Updated this test to check the removing registration code as well. PTAL. Thanks